Rematch/UI.lua

```diff
@@ -9,251 +9,93 @@ local ns = select(2, ...)
     end
 }
 
-local function errorhandler(err)
-    return geterrorhandler()(err)
-end
-
-local function safecall(func, ...)
-    return xpcall(func, errorhandler, ...)
-end
-
-local scriptButtons = setmetatable({}, {
-    __index = function(t, parent)
-        local button = CreateFrame('Button', nil, parent, 'RematchFootnoteButtonTemplate') do
-            if parent.slim then
-                button:SetSize(18, 18)
-            end
-            button:SetPoint('CENTER')
-            button:SetNormalTexture("Interface/AddOns/tdBattlePetScript/Rematch/Textures/ScriptIcon")
-            button:SetPushedTexture("Interface/AddOns/tdBattlePetScript/Rematch/Textures/ScriptIcon")
-            button:SetScript('OnClick', function(button)
-                Addon:OpenScriptEditor(button.key, Rematch:GetTeamTitle(button.key))
-            end)
-            button:SetScript('OnEnter', function(button)
-                GameTooltip:SetOwner(button, 'ANCHOR_RIGHT')
-                GameTooltip:SetText(L.ADDON_NAME)
-                GameTooltip:AddLine(L.WRITE_SCRIPT, HIGHLIGHT_FONT_COLOR:GetRGB())
-                GameTooltip:Show()
-            end)
-            button:SetScript('OnLeave', GameTooltip_Hide)
-        end
-        t[parent] = button
-        return button
-    end
-})
-
 function Addon:OnEnable()
-    local menu = Rematch:GetMenu('TeamMenu')
-    local deleteItem = self:FindMenuItem(menu, DELETE)
-
-    tinsert(menu, 6, teamMenu)
+    -- Add menu to edit script
+    local afterText = Rematch.localization['Set Notes'] -- Use Rematch's locale string.
+    Rematch.menus:AddToMenu('TeamMenu', scriptMenu, afterText)
+    Rematch.menus:AddToMenu('LoadedTeamMenu', scriptMenu, afterText)
+
+    -- Button to indicate a script exists
+    local icon = 'Interface/AddOns/tdBattlePetScript/Rematch/Textures/ScriptIcon'
+    Rematch.badges:RegisterBadge('teams', 'PetBattleScripts', icon, nil, function(teamID)
+        return teamID and self:GetScript(teamID)
+    end)
 
-    -- team delete
+    -- Team is deleted
+    Rematch.events:Register(self, 'REMATCH_TEAM_DELETED', function(self, teamID)
+        self:RemoveScript(teamID)
+    end)
 
-    self:RawHook(deleteItem, 'func', function(obj, key, ...)
-        self.hooks[deleteItem].func(obj, key, ...)
+    -- Maintain sync between script name, and team name
+    Rematch.events:Register(self, 'REMATCH_TEAM_UPDATED', function(self, teamID)
+        local team = Rematch.savedTeams[teamID]
+        local script = self:GetScript(teamID)
 
-        local origAccept = RematchDialog.acceptFunc
-        RematchDialog.acceptFunc = function(...)
-            self:RemoveScript(key)
-            return origAccept(...)
+        if team and script then
+            script:SetName(team.name)
         end
-    end, true)
+    end)
 
-    -- team rename
+    -- TODO: what do we want to do when a team is overwritten? delete the script? keep it? migrate the old script to the new teamID?
```

Comment:
open todo

>  what do we want to do when a team is overwritten? delete the script? keep it? migrate the old script to the new teamID?

Reply:
I don't fully understand, from the callback arguments alone, how to determine what exactly is happening.

Reply:
asked gello in https://www.warcraftpets.com/community/forum/viewtopic.php?p=149579, but it *seems* right